src/unsorted.rs

```diff
@@ -71,6 +81,40 @@ fn mode_on_sorted<T, I>(it: I) -> Option<T>
     mode
 }
 
+fn modes_on_sorted<T, I>(it: I) -> Vec<T>
+        where T: PartialOrd, I: Iterator<Item=T> {
+
+    let mut highest_mode = 1_u32;
+    let mut modes: Vec<u32> = Vec::new();
+    let mut values = Vec::new();
+    let mut count = 0;
+    for x in it {
+        if values.len() == 0 {
+            values.push(x);
+            modes.push(1);
+            continue
+        }
+        if x == values[count] {
+            modes[count] += 1;
+            if highest_mode < modes[count] {
+                highest_mode = modes[count];
+            }
+        } else {
+            values.push(x);
+            modes.push(1);
+            count += 1;
+        }
+    }
+
+
+
+    modes.into_iter()
+            .zip(values)
```

Comment:
Can you use one level of indentation here and for subsequent methods in this chain?